runelite-client/src/main/java/net/runelite/client/plugins/grotesqueguardians/GrotesqueGuardiansPlugin.java

```diff
@@ -50,11 +76,96 @@
 	protected void startUp() throws Exception
 	{
 		overlayManager.add(overlay);
+		overlayManager.add(prayerOverlay);
+		overlayManager.add(textOverlay);
+		dusk = null;
+		prayAgainst = null;
 	}
 
 	@Override
 	protected void shutDown() throws Exception
 	{
 		overlayManager.remove(overlay);
+		overlayManager.remove(textOverlay);
+		overlayManager.remove(prayerOverlay);
+		dusk = null;
+		prayAgainst = null;
+	}
+
+	@Subscribe
+	public void onGameTick(final GameTick event)
+	{
+		final ArrayList<Integer> regions = new ArrayList<Integer>();
+		for (final int intValue : client.getMapRegions())
+		{
+			regions.add(intValue);
+		}
+		if (regions.contains(6727))
+		{
+			dusk = null;
+			inGargs = true;
+			for (final NPC npc : client.getNpcs())
+			{
+				if (npc.getName() != null && npc.getName().contains("Dusk") && !npc.isDead())
+				{
+					dusk = npc;
+				}
+			}
+			if (inGargs && dusk != null)
+			{
+				if (dusk.getId() == 7888)
+				{
+					if (dusk.getAnimation() == DuskAttack.MELEE.getAnimation())
+					{
+						prayAgainst = DuskAttack.MELEE;
+					}
+					else if (dusk.getAnimation() == DuskAttack.RANGE.getAnimation())
+					{
+						prayAgainst = DuskAttack.RANGE;
+					}
+				}
+				else
+					{
+					prayAgainst = null;
+				}
+				if (dusk.getAnimation() == 7802)
+				{
+					needingToRun = true;
+				}
+				else
+					{
+					needingToRun = false;
+				}
+			}
+		}
+		else
+			{
+			inGargs = false;
+			prayAgainst = null;
+			dusk = null;
+		}
 	}
+
+	@Nullable
+	DuskAttack getPrayAgainst()
+	{
+		return prayAgainst;
+	}
+
+	@Nullable
+	NPC getDusk()
```

Comment:
Most of our plugins use lombok getters, its best to stay in that format.

Reply:
not applicable